unitconfig/unitconfig.go

```diff
@@ -189,8 +191,10 @@ func (instance *Instance) UpdateUnitConfig(unitConfig cloudprotocol.UnitConfig)
 	instance.Lock()
 	defer instance.Unlock()
 
-	if err := instance.checkVersion(unitConfig.Version); err != nil {
-		return err
+	if instance.unitConfigError != nil && instance.unitConfig.Version == "" {
```

Comment:
I guess should be OR &
probably better moved unitConfig.Version == ""  check to instance.checkVersion function to avoid duplication

Reply:
This approach was proposed by Oleksandr
The idea is to ignore version check only if smth is wrong with config file and instance unit config version is empty

Reply:
ah. ok